entity-service-api/src/main/proto/org/hypertrace/entity/query/service/v1/entity_query_request.proto

```diff
@@ -35,6 +35,13 @@ message BulkEntityUpdateRequest {
   }
 }
 
+message BulkUpdateEntitiesRequest {
+  // The query defining the results returned after update.
+  // Entity type and filter for update will be fetched from the query.
+  EntityQueryRequest query = 1;
+  repeated UpdateOperation operations = 2;
```

Comment:
Do we lose a way now to specify different types of updates for different entities in a single request?

Reply:
Yes. That can lead to timeout since that would involve making multiple calls to the database.

Moreover, it made sense to return the set of documents (by unioning the ids) in the older way.
But, with different filter criteria, what we return could become confusing if allow different updates based on different criteria.

Reply:
Makes sense. Done.

Reply:
can't add a oneof without breaking compatibility. Maybe we can come up with some clearer naming though (I mean in terms of conveying behavior - not sure if now is the time to start following our general proto best practices in this repo)? The API itself seems fine.

Reply:
oh sorry - I meant on the method name - that it's unclear the difference between bulkUpdate and bulkUpdateEntities. On the best practices, I meant using appropriate casing, removing the streaming result (like line 23 _almost_ did - missed the casing on the method though).

In terms of this field... Putting a query in the update is a little bit weird. Do we intend on limit and offset being enforced? What if they put in a group by?

I think I would probably just give two separate fields - a filter and a selection for the results.

Reply:
> that it's unclear the difference between bulkUpdate and bulkUpdateEntities

Updated it to `BulkUpdateAllMatchingFilter`.

> On the best practices, I meant using appropriate casing, removing the streaming result (like line 23 almost did - missed the casing on the method though).

Renamed the method name to be in title case and removed response streaming.

> Do we intend on limit and offset being enforced?

No. All will be updated at once.

> What if they put in a group by?

Was planning to add a validation in the implementation

> I think I would probably just give two separate fields - a filter and a selection for the results.

Yes. Just taking in the filter now and not returning any results as we can see that's what is being done by the current bulkUpdate API as well.